File: Competitive Programming Exercises/Codeforces/Boy or Girl.cpp
#include <iostream>

using namespace std;

const int AMAX = 100;

bool distinctChars[AMAX];

int main () {
	string s; cin >> s;
	int sLength = (int)s.size();

	for (int i = 0; i < sLength; ++i) {
		distinctChars[s[i] - 'A'] = true;
	}

	int countDistinctChars = 0;
	for (int i = 0; i < AMAX; ++i) {
		if (distinctChars[i]) {
			++countDistinctChars;
		} 
	}

	if (countDistinctChars%2 == 0) {
		cout << "CHAT WITH HER!";
	}
	else {
		cout << "IGNORE HIM!";
	}
}

File: Competitive Programming Exercises/Codeforces/Boy or Girl/Boy or Girl.py
# -*- coding: utf-8 -*-
# @Author: tanht
# @Date:   2020-08-30 22:09:03
# @Last Modified by:   tanht
# @Last Modified time: 2020-08-31 12:57:20

# Source: https://lerner.co.il/2018/06/08/python-parentheses-primer/

# print(["CHAT WITH HER!", "IGNORE HIM!"][len({*input()})%2])
print(["CHAT WITH HER!", "IGNORE HIM!"][len(set(input()))%2])

# {*input()}: create a set from elements of input()